tailnet/conn.go

```diff
@@ -294,6 +294,9 @@ func NewConn(options *Options) (conn *Conn, err error) {
 	}()
 	if server.telemetryStore != nil {
 		server.wireguardEngine.SetNetInfoCallback(func(ni *tailcfg.NetInfo) {
+			server.mutex.Lock()
+			defer server.mutex.Unlock()
+			server.lastNetInfo = ni.Clone()
```

Comment:
To reduce the time we hold the lock for this should probably just be:
```
server.mutex.Lock()
server.lastNetInfo = ni.Clone()
server.mutex.Unlock()
```

In both places